hkmc2/shared/src/main/scala/hkmc2/semantics/Resolver.scala

```diff
@@ -831,18 +832,93 @@ class Resolver(tl: TraceLogger)
         // the type signature should be present.
         lastWords(s"No type signature for contextual parameter ${p.showDbg} at ${p.toLoc}")
   
-  def resolveParam(p: Param)(using ictx: ICtx): Unit =
+  def traverseParam(p: Param)(using ictx: ICtx): Unit =
     log(s"Resolving parameter ${p.showDbg}")
     if p.modulefulness.modified then
       if p.sign.isEmpty then
         raise(ErrorReport(msg"Module parameter must have explicit type." -> p.sym.toLoc :: Nil))
     p.sign.foreach:
-      traverse(_, 
+      traverseType(_, 
         expect = if p.modulefulness.modified
           then Module(S(msg"Module parameter must have a module type."))
           else NonModule(S(msg"Non-module parameter must have a non-module type.")),
       )
   
+  def traverseType(t: Term, expect: Expect, inTyAppPrefix: Bool = false)(using ictx: ICtx): Unit =
+  trace(s"Traversing type ${t}, expecting ${expect}"):
+    def checkTypeArity(sym: FieldSymbol): Unit =
+      if inTyAppPrefix && !t.isInstanceOf[Term.TyApp] then
```

Comment:
Why would you abort the checks in this case? Can you documemt the reason for the `!t.isInstanceOf[Term.TyApp]` check? @FlandiaYingman